packages/elements/src/tree/helpers/filter.ts

```diff
@@ -0,0 +1,34 @@
+import { Tree, TreeDataItem } from '../index';
+import { TreeFilter } from './types';
+
+/**
+ * Default filter used by tree
+ * @param el Tree instance to filter
+ * @returns Filter accepting an item
+ */
+export const defaultFilter = <T extends TreeDataItem>(el: Tree<T>): TreeFilter<T> => {
+  // reference query string for validating queryRegExp cache state
+  let query = '';
+  // cache RegExp
+  let queryRegExp: RegExp | undefined;
+
+  // Get current RegExp, or renew if out of date
+  // this is fetched on demand by filter/renderer
+  // only created once per query
+  const getRegularExpressionOfQuery = (): RegExp => {
+    if (el.query !== query || !queryRegExp) {
+      query = el.query || '';
+      queryRegExp = new RegExp(query.replace(/(\W)/g, '\\$1'), 'i');
+    }
+    return queryRegExp;
+  };
+
+  // return scoped custom filter
+  return (item): boolean => {
+    const regex = getRegularExpressionOfQuery();
+    const result = regex.test(String((item as unknown as TreeDataItem).label));
```

Comment:
String(null) === 'null' and String(undefined) === 'undefined'. It does not make sense to filter on null or undefined. I label is not present, it should tested agains empty string ('') or in other words simply excluded from the result.